src/__swaps__/screens/Swap/providers/SyncSwapStateAndSharedValues.tsx

```diff
@@ -128,7 +127,12 @@ export function SyncGasStateToSharedValues() {
 
   useEffect(() => {
     hasEnoughFundsForGas.value = undefined;
-    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || !userNativeNetworkAsset) return;
+    if (!gasSettings || !estimatedGasLimit || !quote || 'error' in quote || isLoadingNativeNetworkAsset) return;
+
+    if (!userNativeNetworkAsset) {
+      hasEnoughFundsForGas.value = false;
+      return;
+    }
```

Comment:
tested with an account that didn't have any gas token, found out we have to differentiate between loading the `userNativeNetworkAsset` and not having it :)